velox/exec/WindowPartition.cpp

```diff
@@ -398,10 +399,12 @@ void WindowPartition::updateKRangeFrameBounds(
       // [0, currentRow] are examined. For following bounds, rows between
       // [currentRow, numRows()) are checked.
       if (isPreceding) {
-        start = 0;
+        start = lastFoundIndex == -1 ? 0 : lastFoundIndex;
```

Comment:
@PHILO-HE : My hunch is that you would need more than this change to use lastFoundIndex. You could check the value of the frameColumn at this row with that of lastFoundIndex to use this optimization. If the frame bound value does not follow the sliding pattern that this optimization is targeted for then its not applicable.